tmcprototype/SubarrayNode/SubarrayNode/SubarrayNode.py

```diff
@@ -124,26 +124,50 @@ def is_EndScan_allowed(self):
         """
         try:
+            # Allocation success and failure lists
+            allocation_success = []
+            allocation_failure = []
             for leafId in range(0, len(argin)):
                 if type(float(argin[leafId])) == float:
                     pass
             for leafId in range(0, len(argin)):
-                self._dish_leaf_node_group.add(self.DishLeafNodePrefix +  argin[leafId])
-                devProxy = tango.DeviceProxy(self.DishLeafNodePrefix + argin[leafId])
-                self._dish_leaf_node_proxy.append(devProxy)
-                self._event_id = devProxy.subscribe_event(CONST.EVT_DISH_HEALTH_STATE,
-                                                          tango.EventType.CHANGE_EVENT,
-                                                          self.setHealth,
-                                                          stateless=True)
-                self.testDeviceVsEventID[devProxy] = self._event_id
-                self._health_event_id.append(self._event_id)
-                self._receptor_id_list.append(int(argin[leafId]))
-                self.dishHealthStateMap[devProxy] = -1
+                try:
+                    self._dish_leaf_node_group.add(self.DishLeafNodePrefix +  argin[leafId])
+                    devProxy = tango.DeviceProxy(self.DishLeafNodePrefix + argin[leafId])
+                    self._dish_leaf_node_proxy.append(devProxy)
+                    # Update the list allocation_success with the dishes allocated successfully to subarray
+                    allocation_success.append(argin[leafId])
+                    self._event_id = devProxy.subscribe_event(CONST.EVT_DISH_HEALTH_STATE,
+                                                              tango.EventType.CHANGE_EVENT,
+                                                              self.setHealth,
+                                                              stateless=True)
+                    self.testDeviceVsEventID[devProxy] = self._event_id
+                    self._health_event_id.append(self._event_id)
+                    self._receptor_id_list.append(int(argin[leafId]))
+                    self.dishHealthStateMap[devProxy] = -1
+
+                except Exception as except_occured:
+                    allocation_failure.append(argin[leafId])
+                    # Exception Logic to remove Id from subarray group
+                    group_dishes = self._dish_leaf_node_group.get_device_list()
+                    if group_dishes.contains(self.DishLeafNodePrefix +  argin[leafId]):
+                        self._dish_leaf_node_group.remove(self.DishLeafNodePrefix + argin[leafId])
+                    # unsubscribe event
+                    if self.testDeviceVsEventID[devProxy]:
+                        devProxy.unsubscribe_event(self.testDeviceVsEventID[devProxy])
+
             print(CONST.STR_TEST_DEV_VS_EVT_ID, self.testDeviceVsEventID)
```

Comment:
are these print statements required?

Reply:
These print statements are required. Removed all the unnecessary print statements and commented code.